common/persistence/cassandra/queue_v2_store.go

```diff
@@ -474,3 +476,32 @@ func (s *queueV2Store) getMaxMessageID(ctx context.Context, queueType persistenc
 	}
 	return maxMessageID, true, nil
 }
+
+func (s *queueV2Store) ListQueues(
+	ctx context.Context,
+	request *persistence.InternalListQueuesRequest,
+) (*persistence.InternalListQueuesResponse, error) {
+	if request.PageSize <= 0 {
+		return nil, persistence.ErrNonPositiveListQueuesPageSize
+	}
```

Comment:
I think we should move this up to the manager layer so that we don't duplicate it across DBs